smarts/env/hiway_env.py

```diff
@@ -129,22 +129,62 @@ def __init__(
         if visdom:
             visdom_client = VisdomClient()
 
-        self._smarts = SMARTS(
-            agent_interfaces=agent_interfaces,
-            traffic_sim=SumoTrafficSimulation(
+        is_opendrive = HiWayEnv.check_scenario_versions(scenarios)
+        traffic_sim = None
+        if is_opendrive:
+            # We currently don't support the Native SUMO Traffic Provider or Social Agents for OpenDRIVE maps
+            if zoo_addrs:
+                warnings.warn("`zoo_addrs` should not be used with OpenDRIVE scenarios")
+                zoo_addrs = None
+            pass
+        else:
+            from smarts.core.sumo_traffic_simulation import SumoTrafficSimulation
+
+            traffic_sim = SumoTrafficSimulation(
                 headless=sumo_headless,
                 time_resolution=fixed_timestep_sec,
                 num_external_sumo_clients=num_external_sumo_clients,
                 sumo_port=sumo_port,
                 auto_start=sumo_auto_start,
                 endless_traffic=endless_traffic,
-            ),
+            )
+            zoo_addrs = zoo_addrs
+
+        self._smarts = SMARTS(
+            agent_interfaces=agent_interfaces,
+            traffic_sim=traffic_sim,
             envision=envision_client,
             visdom=visdom_client,
             fixed_timestep_sec=fixed_timestep_sec,
             zoo_addrs=zoo_addrs,
         )
 
+    @staticmethod
+    def check_scenario_versions(scenarios):
+        num_sumo = 0
+        num_opendrive = 0
+        scenario_list = Scenario.get_scenario_list(scenarios)
+        for scenario_root in scenario_list:
+            if os.path.exists(os.path.join(scenario_root, "map.net.xml")):
+                num_sumo += 1
+            elif os.path.exists(os.path.join(scenario_root, "map.xodr")):
+                num_opendrive += 1
+            else:
+                raise FileNotFoundError(
+                    f"Unable to find map in map_source={scenario_root}."
+                )
+
+        try:
+            assert (num_sumo == 0 and num_opendrive == len(scenarios)) or (
+                num_sumo == len(scenarios) and num_opendrive == 0
+            )
+        except AssertionError:
+            raise AssertionError(
+                "All scenarios passed as a parameters for the gym environment need to be of the same version (OpenDRIVE or SUMO)."
+            )
```

Comment:
I guess this may be moot if you make changes like I suggested above, but here I'm wondering, why not just:
```python
if (num_sumo > 0 and num_opendrive > 0) or (num_sumo + num_opendrive != len(scenarios)):
    raise AssertionError( "All scenarios passed as a parameters for the gym environment need to be of the same version (OpenDRIVE or SUMO).")
```